sdk/environments/environment.go

```diff
@@ -160,7 +160,7 @@ func NewApiEndpoint(name, endpoint string, appId *string) *ApiEndpoint {
 	}
 }
 
-func (e *ApiEndpoint) withResourceIdentifier(identifier string) *ApiEndpoint {
+func (e *ApiEndpoint) WithResourceIdentifier(identifier string) Api {
 	e.resourceIdentifier = pointer.To(identifier)
 	return e
 }
```

Comment:
Since there's talk of overriding the token below on a per-Storage Account basis - perhaps this should return a new instance of `Api` rather than mutating the existing object, to ensure this is a clean-copy?